pistachio: add 5.15 as testing kernel

PolynomialDivision opened the pull request:
currently fails because of:
```
rm -f /home/nick/openwrt/build_dir/target-mipsel_24kc+24kf_musl/linux-pistachio/img_creator-ci40-kernel.itb
cp /home/nick/openwrt/build_dir/target-mipsel_24kc+24kf_musl/linux-pistachio/vmlinux /home/nick/openwrt/build_dir/target-mipsel_24kc+24kf_musl/linux-pistachio/img_creator-ci40-kernel.itb
gzip -f -9n -c /home/nick/openwrt/build_dir/target-mipsel_24kc+24kf_musl/linux-pistachio/img_creator-ci40-kernel.itb  > /home/nick/openwrt/build_dir/target-mipsel_24kc+24kf_musl/linux-pistachio/img_creator-ci40-kernel.itb.new
/home/nick/openwrt/scripts/mkits.sh -D img_creator-ci40 -o /home/nick/openwrt/build_dir/target-mipsel_24kc+24kf_musl/linux-pistachio/img_creator-ci40-kernel.itb.its -k /home/nick/openwrt/build_dir/target-mipsel_24kc+24kf_musl/linux-pistachio/img_creator-ci40-kernel.itb -C gzip   -d /home/nick/openwrt/build_dir/target-mipsel_24kc+24kf_musl/linux-pistachio/linux-5.15.69/arch/mips/boot/dts/img/pistachio_marduk.dtb   -a 0x80400000 -e 0x80400000    -c config@1 -A mips -v 5.15.69
PATH=/home/nick/openwrt/build_dir/target-mipsel_24kc+24kf_musl/linux-pistachio/linux-5.15.69/scripts/dtc:/home/nick/openwrt/staging_dir/toolchain-mipsel_24kc+24kf_gcc-11.3.0_musl/bin:/home/nick/openwrt/staging_dir/toolchain-mipsel_24kc+24kf_gcc-11.3.0_musl/bin:/home/nick/openwrt/staging_dir/toolchain-mipsel_24kc+24kf_gcc-11.3.0_musl/bin:/home/nick/openwrt/staging_dir/toolchain-mipsel_24kc+24kf_gcc-11.3.0_musl/bin:/home/nick/openwrt/staging_dir/host/bin:/home/nick/openwrt/staging_dir/host/bin:/usr/local/sbin:/usr/local/bin:/usr/bin:/usr/bin/site_perl:/usr/bin/vendor_perl:/usr/bin/core_perl mkimage  -f /home/nick/openwrt/build_dir/target-mipsel_24kc+24kf_musl/linux-pistachio/img_creator-ci40-kernel.itb.its /home/nick/openwrt/build_dir/target-mipsel_24kc+24kf_musl/linux-pistachio/img_creator-ci40-kernel.itb.new
FATAL ERROR: Couldn't open "/home/nick/openwrt/build_dir/target-mipsel_24kc+24kf_musl/linux-pistachio/linux-5.15.69/arch/mips/boot/dts/img/pistachio_marduk.dtb": No such file or directory
mkimage: Can't open /home/nick/openwrt/build_dir/target-mipsel_24kc+24kf_musl/linux-pistachio/img_creator-ci40-kernel.itb.new.tmp: No such file or directory
Error: Bad parameters for FIT image type
Usage: mkimage [-T type] -l image
```

PolynomialDivision committed:
pistachio: 5:15: copy config and patch from 5.10

Copy config and patch from kernel 5.10 to kernel 5.15.

Signed-off-by: Nick Hainke <vincent@systemli.org>
[Updated the copy]
Signed-off-by: Hauke Mehrtens <hauke@hauke-m.de>

PolynomialDivision committed:
pistachio: add 5.15 testing kernel

Manually refreshed:
- 903-MIPS-DTS-img-marduk-Add-NXP-SC16IS752IPW.patch
- 905-MIPS-DTS-img-marduk-Add-led-aliases.patch (led node was renamed to led-1)

Automatically refreshed:
- 109-MIPS-DTS-img-marduk-switch-mmc-to-1-bit-mode.patch
- 401-mtd-nor-support-mtd-name-from-device-tree.patch
- 901-MIPS-DTS-img-marduk-Add-SPI-NAND-flash.patch
- 902-MIPS-DTS-img-marduk-Add-Cascoda-CA8210-6LoWPAN.patch
- 904-MIPS-DTS-img-marduk-Add-partition-name.patch

Enable testing kernel.

Co-Developed-by: Hauke Mehrtens <hauke@hauke-m.de>
Tested-by: Hauke Mehrtens <hauke@hauke-m.de>
Signed-off-by: Nick Hainke <vincent@systemli.org>

hauke commented on target/linux/pistachio/config-5.15:
```diff
@@ -63,25 +68,34 @@ CONFIG_DMA_OF=y
+# CONFIG_FIT_IMAGE_FDT_BOSTON is not set
+# CONFIG_FIT_IMAGE_FDT_JAGUAR2 is not set
+# CONFIG_FIT_IMAGE_FDT_LUTON is not set
+# CONFIG_FIT_IMAGE_FDT_MARDUK is not set
```
You have to activate `CONFIG_FIT_IMAGE_FDT_MARDUK` to build the dtb file. This should fix your build problem.

PolynomialDivision replied:
Thanks! :)

hauke commented on target/linux/pistachio/config-5.15:
```diff
@@ -1,33 +1,39 @@
+# CONFIG_BOARD_INGENIC is not set
 CONFIG_BOARD_SCACHE=y
-CONFIG_BOOT_ELF32=y
+CONFIG_BUILTIN_DTB=y
```
Why do you activate `CONFIG_BUILTIN_DTB` ? I think the dtb is in a separate file on the sd card.

PolynomialDivision replied:
Was automatically activated by make kernel_oldconfig. I looked at some other targets and `malta` and `octeon` have it also activated.

hauke commented on target/linux/pistachio/config-5.15:
```diff
@@ -187,8 +204,9 @@ CONFIG_NEED_DMA_MAP_STATE=y
 CONFIG_NET_NS=y
 CONFIG_NET_PTP_CLASSIFY=y
+CONFIG_NET_SELFTESTS=y
 CONFIG_NLS=y
```
I think we should remove these options:
````
CONFIG_NET_FLOW_LIMIT=y
CONFIG_NET_NS=y
CONFIG_NET_PTP_CLASSIFY=y
CONFIG_NET_SELFTESTS=y
CONFIG_NLS=y
```
in a separate patch .

PolynomialDivision replied:
Why?

hauke replied:
These configurations are not target specific and should use the settings from the generic configuration. This was already a problem in the configuration for kernel 5.10.

PolynomialDivision replied:
Okay, so doing a tree-wide patch for all targets moving those options to common.

hauke commented on target/linux/pistachio/config-5.15:
```diff
@@ -1,33 +1,39 @@
 CONFIG_CLOCKSOURCE_WATCHDOG=y
 CONFIG_CLONE_BACKWARDS=y
 CONFIG_COMMON_CLK=y
-# CONFIG_COMMON_CLK_BOSTON is not set
```
Please activate `CONFIG_COMMON_CLK_PISTACHIO`

PolynomialDivision replied:
Done. :)

Ansuel commented:
wth happen to the labeller action WOW

robimarko commented:
Labeler got a bit trigger happy

xabolcs commented:
Nothing wrong with labeller, just the ~ treewide commit 98c299634f68514662e1523a1dfa209585f21311 was added.

Ansuel commented:
For this kind of change we should just add the treewide tag

Il Lun 3 Ott 2022, 12:33 Szabolcs Hubai ***@***.***> ha
scritto:

> Nothing wrong with labeller, just the ~ treewide commit 98c2996
> <https://github.com/openwrt/openwrt/commit/98c299634f68514662e1523a1dfa209585f21311>
> was added.
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/openwrt/openwrt/pull/10841#issuecomment-1265247786>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AE2ZMQVDP7EYDHZ4UCKN7KTWBKZAZANCNFSM6AAAAAAQX2J4ZQ>
> .
> You are receiving this because you commented.Message ID:
> ***@***.***>
>

hauke commented:
I did not expect that so many targets were affected by this. probably better to do the tree wide change in a separate pull request.

Did someone test this image with kernel 5.15 on a pistachio board? I have one, but haven't tried yet.

hauke commented:
The kernel 5.15 image is not booting up for me. 
````
U-Boot SPL 2015.10 (Dec 09 2016 - 10:51:44)


U-Boot 2015.10 (Dec 09 2016 - 10:51:44 +0000), Build: jenkins-CreatorDev-openwrt-ci40-128

MIPS(interAptiv): IMG Pistachio 546MHz.
Model: IMG Marduk
DRAM:  256 MiB
NAND:  512 MiB
MMC:   Synopsys Mobile storage: 0
SF: Detected W25Q16CL with page size 256 Bytes, erase size 4 KiB, total 2 MiB
In:    uart@18101500
Out:   uart@18101500
Err:   uart@18101500
TPM:   Infineon (1.2)
Net:   eth0: ethernet@18140000
Warning: Bootlimit (5) exceeded. Using altbootcmd.
Hit any key to stop autoboot:  0 
SF: Detected W25Q16CL with page size 256 Bytes, erase size 4 KiB, total 2 MiB
Boot failure detected, switching to firmware0
Saving Environment to SPI Flash...
SF: Detected W25Q16CL with page size 256 Bytes, erase size 4 KiB, total 2 MiB
Erasing SPI flash...Writing to SPI flash...done
Attempting to boot from firmware0
UBI: attaching mtd3 to ubi0
UBI: scanning is finished
UBI: attached mtd3 (name "mtd=0", size 256 MiB) to ubi0
UBI: PEB size: 262144 bytes (256 KiB), LEB size: 253952 bytes
UBI: min./max. I/O unit sizes: 4096/4096, sub-page size 4096
UBI: VID header offset: 4096 (aligned 4096), data offset: 8192
UBI: good PEBs: 1024, bad PEBs: 0, corrupted PEBs: 0
UBI: user volume: 3, internal volumes: 1, max. volumes count: 128
UBI: max/mean erase counter: 77/39, WL threshold: 4096, image sequence number: 1496738702
UBI: available PEBs: 0, total reserved PEBs: 1024, PEBs reserved for bad PEB handling: 40
Loading kernel from volume...
Read 0 bytes from volume kernel to e000000
No size specified -> Using max size (4825088)
## Loading kernel from FIT Image at 0e000000 ...
   Using 'config@1' configuration
   Trying 'kernel-1' kernel subimage
     Description:  MIPS OpenWrt Linux-5.15.79
     Type:         Kernel Image
     Compression:  gzip compressed
     Data Start:   0x0e0000e4
     Data Size:    4657902 Bytes = 4.4 MiB
     Architecture: MIPS
     OS:           Linux
     Load Address: 0x81000000
     Entry Point:  0x81000000
     Hash algo:    crc32
     Hash value:   d4127928
     Hash algo:    sha1
     Hash value:   806f1d7b9df15eecbbdf172924e2341d4085c8a5
## Loading fdt from FIT Image at 0e000000 ...
   Using 'config@1' configuration
   Trying 'fdt-1' fdt subimage
     Description:  MIPS OpenWrt img_creator-ci40 device tree blob
     Type:         Flat Device Tree
     Compression:  uncompressed
     Data Start:   0x0e471514
     Data Size:    18190 Bytes = 17.8 KiB
     Architecture: MIPS
     Hash algo:    crc32
     Hash value:   6a432246
     Hash algo:    sha1
     Hash value:   6b85a3143cea26de59500691b7e74fbce45e327c
   Booting using the fdt blob at 0xe471514
   Uncompressing Kernel Image ... OK
WARNING: can't find wifi0 alias
U-Boot SPL 2015.10 (Dec 09 2016 - 10:51:44)
````
I already changed the KERNEL_LOADADDR which changes in the kernel build:
https://github.com/torvalds/linux/blob/v5.15/arch/mips/generic/Platform#L15
https://github.com/torvalds/linux/blob/v5.10/arch/mips/pistachio/Platform#L6


While trying kernel 5.10 I found this bug: https://www.spinics.net/lists/linux-wpan/msg07472.html

PolynomialDivision commented:
> While trying kernel 5.10 I found this bug: https://www.spinics.net/lists/linux-wpan/msg07472.html

Your patch is applied in next. https://www.spinics.net/lists/linux-wpan/msg07486.html
Sadly, I don't have a device. :/ So I can not further debug this.

PolynomialDivision commented:
@hauke Do you think that you can manage to find the underlying bug? Sadly, I don't own such a device I could use for testing.

hauke commented:
When I find some time I will look into this again. Please keep the pull request open for now.
Maybe we are missing a magic kernel option. ;-)

hauke commented:
This is needed to make it boot up with kernel 5.15:
````
diff --git a/target/linux/pistachio/config-5.15 b/target/linux/pistachio/config-5.15
index 54c93b8605e7..c303e99987a1 100644
--- a/target/linux/pistachio/config-5.15
+++ b/target/linux/pistachio/config-5.15
@@ -8,7 +8,7 @@ CONFIG_BLK_DEV_LOOP=y
 CONFIG_BLK_DEV_SD=y
 # CONFIG_BOARD_INGENIC is not set
 CONFIG_BOARD_SCACHE=y
-# CONFIG_BUILTIN_DTB is not set
+CONFIG_BUILTIN_DTB=y
 CONFIG_CEVT_R4K=y
 CONFIG_CLKSRC_MIPS_GIC=y
 CONFIG_CLKSRC_PISTACHIO=y
@@ -27,6 +27,7 @@ CONFIG_CPU_HAS_SYNC=y
 CONFIG_CPU_IDLE=y
 CONFIG_CPU_IDLE_GOV_LADDER=y
 CONFIG_CPU_IDLE_GOV_MENU=y
+CONFIG_CPU_LITTLE_ENDIAN=y
 # CONFIG_CPU_MICROMIPS is not set
 CONFIG_CPU_MIPS32=y
 # CONFIG_CPU_MIPS32_R1 is not set
diff --git a/target/linux/pistachio/image/Makefile b/target/linux/pistachio/image/Makefile
index 2e15005ac230..9c0e9da91f52 100644
--- a/target/linux/pistachio/image/Makefile
+++ b/target/linux/pistachio/image/Makefile
@@ -5,7 +5,11 @@
 include $(TOPDIR)/rules.mk
 include $(INCLUDE_DIR)/image.mk
 
-KERNEL_LOADADDR := 0x80400000
+ifdef CONFIG_LINUX_5_10
+  KERNEL_LOADADDR := 0x80400000
+else
+  KERNEL_LOADADDR := 0x80100000
+endif
 
 define Device/Default
   PROFILES := Default
````
Please integrate it into your 2. patch. 
mainly the `CONFIG_CPU_LITTLE_ENDIAN` was missing.

It boots up like this:
````
UBI: attaching mtd3 to ubi0
UBI: scanning is finished
UBI: attached mtd3 (name "mtd=0", size 256 MiB) to ubi0
UBI: PEB size: 262144 bytes (256 KiB), LEB size: 253952 bytes
UBI: min./max. I/O unit sizes: 4096/4096, sub-page size 4096
UBI: VID header offset: 4096 (aligned 4096), data offset: 8192
UBI: good PEBs: 1024, bad PEBs: 0, corrupted PEBs: 0
UBI: user volume: 3, internal volumes: 1, max. volumes count: 128
UBI: max/mean erase counter: 89/45, WL threshold: 4096, image sequence number: 516423239
UBI: available PEBs: 0, total reserved PEBs: 1024, PEBs reserved for bad PEB handling: 40
Loading kernel from volume...
Read 0 bytes from volume kernel to e000000
No size specified -> Using max size (4825088)
## Loading kernel from FIT Image at 0e000000 ...
   Using 'config@1' configuration
   Trying 'kernel-1' kernel subimage
     Description:  MIPS OpenWrt Linux-5.15.89
     Type:         Kernel Image
     Compression:  gzip compressed
     Data Start:   0x0e0000e4
     Data Size:    4626064 Bytes = 4.4 MiB
     Architecture: MIPS
     OS:           Linux
     Load Address: 0x80100000
     Entry Point:  0x80100000
     Hash algo:    crc32
     Hash value:   31f1959e
     Hash algo:    sha1
     Hash value:   c60df915b471ca7172227493153a60b2e5c5d92c
## Loading fdt from FIT Image at 0e000000 ...
   Using 'config@1' configuration
   Trying 'fdt-1' fdt subimage
     Description:  MIPS OpenWrt img_creator-ci40 device tree blob
     Type:         Flat Device Tree
     Compression:  uncompressed
     Data Start:   0x0e4698b4
     Data Size:    18190 Bytes = 17.8 KiB
     Architecture: MIPS
     Hash algo:    crc32
     Hash value:   6a432246
     Hash algo:    sha1
     Hash value:   6b85a3143cea26de59500691b7e74fbce45e327c
   Booting using the fdt blob at 0xe4698b4
   Uncompressing Kernel Image ... OK
WARNING: can't find wifi0 alias
[    0.000000] Linux version 5.15.89 (hauke@hauke-t480) (mipsel-openwrt-linux-musl-gcc (OpenWrt GCC 12.2.0 r21878-6e3acebe35cf) 12.2.0, GNU ld (GNU Binutils) 2.39) #0 SMP Sun Jan 22 16:45:41 2023
[    0.000000] CPU0 revision is: 0001a120 (MIPS interAptiv (multi))
[    0.000000] FPU revision is: 0173a000
[    0.000000] MIPS: machine is IMG Marduk (Creator Ci40)
[    0.000000] earlycon: uart8250 at MMIO32 0x18101500 (options '115200')
[    0.000000] printk: bootconsole [uart8250] enabled
[    0.000000] Initrd not found or empty - disabling initrd
[    0.000000] VPE topology {2,2} total 4
[    0.000000] Primary instruction cache 32kB, VIPT, 4-way, linesize 32 bytes.
[    0.000000] Primary data cache 32kB, 4-way, PIPT, no aliases, linesize 32 bytes
[    0.000000] MIPS secondary cache 512kB, 8-way, linesize 32 bytes.
[    0.000000] Zone ranges:
[    0.000000]   Normal   [mem 0x0000000000000000-0x000000000fffffff]
[    0.000000] Movable zone start for each node
[    0.000000] Early memory node ranges
[    0.000000]   node   0: [mem 0x0000000000000000-0x000000000fffffff]
[    0.000000] Initmem setup node 0 [mem 0x0000000000000000-0x000000000fffffff]
[    0.000000] percpu: Embedded 12 pages/cpu s17552 r8192 d23408 u49152
[    0.000000] Built 1 zonelists, mobility grouping on.  Total pages: 64960
[    0.000000] Kernel command line: console=ttyS1,115200n8 earlycon=uart8250,mmio32,0x18101500,115200 ubi.mtd=firmware0 rootwait ro lpj=723968 mtdparts=spi-nand:256M(firmware0),256M(firmware1);spi-nor:1536k(uboot),8k(data-ro),8k(uEnv),496k(data-rw) panic=2
[    0.000000] Dentry cache hash table entries: 32768 (order: 5, 131072 bytes, linear)
[    0.000000] Inode-cache hash table entries: 16384 (order: 4, 65536 bytes, linear)
[    0.000000] Writing ErrCtl register=00000000
[    0.000000] Readback ErrCtl register=00000000
[    0.000000] mem auto-init: stack:off, heap alloc:off, heap free:off
[    0.000000] Memory: 245980K/262144K available (7992K kernel code, 1061K rwdata, 1636K rodata, 244K init, 349K bss, 16164K reserved, 0K cma-reserved)
[    0.000000] SLUB: HWalign=128, Order=0-3, MinObjects=0, CPUs=4, Nodes=1
[    0.000000] rcu: Hierarchical RCU implementation.
[    0.000000]  Tracing variant of Tasks RCU enabled.
[    0.000000] rcu: RCU calculated value of scheduler-enlistment delay is 10 jiffies.
[    0.000000] NR_IRQS: 256
[    0.000000] sched_clock: 32 bits at 350MHz, resolution 2ns, wraps every 6135667710ns
[    0.000000] clocksource: gptimer: mask: 0xffffffff max_cycles: 0xffffffff, max_idle_ns: 5460744132 ns
[    0.000000] clocksource: GIC: mask: 0xffffffffffffffff max_cycles: 0x7deca086e2, max_idle_ns: 440795208515 ns
[    0.000002] sched_clock: 64 bits at 546MHz, resolution 1ns, wraps every 2199023255551ns
[    0.009055] clocksource: MIPS: mask: 0xffffffff max_cycles: 0xffffffff, max_idle_ns: 7000954018 ns
[    0.019477] Calibrating delay loop (skipped) preset value.. 144.79 BogoMIPS (lpj=723968)
[    0.028566] pid_max: default: 32768 minimum: 301
[    0.034121] Mount-cache hash table entries: 1024 (order: 0, 4096 bytes, linear)
[    0.042324] Mountpoint-cache hash table entries: 1024 (order: 0, 4096 bytes, linear)
[    0.055798] rcu: Hierarchical SRCU implementation.
[    0.061731] dyndbg: Ignore empty _ddebug table in a CONFIG_DYNAMIC_DEBUG_CORE build
[    0.070874] smp: Bringing up secondary CPUs ...
[    0.077138] Primary instruction cache 32kB, VIPT, 4-way, linesize 32 bytes.
[    0.077161] Primary data cache 32kB, 4-way, PIPT, no aliases, linesize 32 bytes
[    0.077185] MIPS secondary cache 512kB, 8-way, linesize 32 bytes.
[    0.077302] CPU1 revision is: 0001a120 (MIPS interAptiv (multi))
[    0.077323] FPU revision is: 0173a000
[    0.077479] Synchronize counters for CPU 1: done.
[    0.120059] Primary instruction cache 32kB, VIPT, 4-way, linesize 32 bytes.
[    0.120082] Primary data cache 32kB, 4-way, PIPT, no aliases, linesize 32 bytes
[    0.120098] MIPS secondary cache 512kB, 8-way, linesize 32 bytes.
[    0.120151] CPU2 revision is: 0001a120 (MIPS interAptiv (multi))
[    0.120171] FPU revision is: 0173a000
[    0.122010] Synchronize counters for CPU 2: done.
[    0.162060] Primary instruction cache 32kB, VIPT, 4-way, linesize 32 bytes.
[    0.162081] Primary data cache 32kB, 4-way, PIPT, no aliases, linesize 32 bytes
[    0.162096] MIPS secondary cache 512kB, 8-way, linesize 32 bytes.
[    0.162162] CPU3 revision is: 0001a120 (MIPS interAptiv (multi))
[    0.162182] FPU revision is: 0173a000
[    0.162244] Synchronize counters for CPU 3: done.
[    0.201298] smp: Brought up 1 node, 4 CPUs
[    0.214825] clocksource: jiffies: mask: 0xffffffff max_cycles: 0xffffffff, max_idle_ns: 19112604462750000 ns
[    0.226001] futex hash table entries: 1024 (order: 5, 131072 bytes, linear)
[    0.234251] pinctrl core: initialized pinctrl subsystem
[    0.242607] NET: Registered PF_NETLINK/PF_ROUTE protocol family
[    0.250754] cpuidle: using governor ladder
[    0.299908] SCSI subsystem initialized
[    0.304594] usbcore: registered new interface driver usbfs
[    0.310835] usbcore: registered new interface driver hub
[    0.316911] usbcore: registered new device driver usb
[    0.322657] pps_core: LinuxPPS API ver. 1 registered
[    0.328296] pps_core: Software ver. 5.3.6 - Copyright 2005-2007 Rodolfo Giometti <giometti@linux.it>
[    0.338587] PTP clock support registered
[    0.345317] clocksource: Switched to clocksource gptimer
[    0.352695] NET: Registered PF_INET protocol family
[    0.358578] IP idents hash table entries: 4096 (order: 3, 32768 bytes, linear)
[    0.367909] tcp_listen_portaddr_hash hash table entries: 512 (order: 0, 6144 bytes, linear)
[    0.377370] Table-perturb hash table entries: 65536 (order: 6, 262144 bytes, linear)
[    0.386083] TCP established hash table entries: 2048 (order: 1, 8192 bytes, linear)
[    0.394719] TCP bind hash table entries: 2048 (order: 2, 16384 bytes, linear)
[    0.402826] TCP: Hash tables configured (established 2048 bind 2048)
[    0.410176] UDP hash table entries: 256 (order: 1, 8192 bytes, linear)
[    0.417591] UDP-Lite hash table entries: 256 (order: 1, 8192 bytes, linear)
[    0.425808] NET: Registered PF_UNIX/PF_LOCAL protocol family
[    0.434370] workingset: timestamp_bits=14 max_order=16 bucket_order=2
[    0.449590] squashfs: version 4.0 (2009/01/31) Phillip Lougher
[    0.456240] jffs2: version 2.2 (NAND) (SUMMARY) (LZMA) (RTIME) (CMODE_PRIORITY) (c) 2001-2006 Red Hat, Inc.
[    0.476959] img-mdc-dma 18143000.dma-controller: MDC with 12 channels and 2 threads
[    0.486576] Serial: 8250/16550 driver, 2 ports, IRQ sharing disabled
[    0.506906] 18101400.uart: ttyS0 at MMIO 0x18101400 (irq = 22, base_baud = 115740) is a 16550A
[    0.528916] 18101500.uart: ttyS1 at MMIO 0x18101500 (irq = 23, base_baud = 115740) is a 16550A
[    0.538691] printk: console [ttyS1] enabled
[    0.538691] printk: console [ttyS1] enabled
[    0.548035] printk: bootconsole [uart8250] disabled
[    0.548035] printk: bootconsole [uart8250] disabled
[    0.572247] loop: module loaded
[    0.578291] ca8210@0 enforce active low on chipselect handle
[    0.584720] sc16is752@1 enforce active low on chipselect handle
[    0.595890] spi0.1: ttySC1 at I/O 0x1 (irq = 48, base_baud = 1000000) is a SC16IS752
[    0.607078] flash@0 enforce active low on chipselect handle
[    0.613401] flash@1 enforce active low on chipselect handle
[    0.621581] spi-nor spi1.0: s25fl016k (2048 Kbytes)
[    0.627202] 4 cmdlinepart partitions found on MTD device spi-nor
[    0.633903] Creating 4 MTD partitions on "spi-nor":
[    0.639387] 0x000000000000-0x000000180000 : "uboot"
[    0.646476] 0x000000180000-0x000000182000 : "data-ro"
[    0.653189] 0x000000182000-0x000000184000 : "uEnv"
[    0.659702] 0x000000184000-0x000000200000 : "data-rw"
[    0.670225] spi-nand spi1.1: GigaDevice SPI NAND was found.
[    0.676587] spi-nand spi1.1: 512 MiB, block size: 256 KiB, page size: 4096, OOB size: 256
[    0.685970] 2 cmdlinepart partitions found on MTD device spi-nand
[    0.692769] Creating 2 MTD partitions on "spi-nand":
[    0.698348] 0x000000000000-0x000010000000 : "firmware0"
[    1.006901] 0x000010000000-0x000020000000 : "firmware1"
[    1.321876] stmmaceth 18140000.ethernet: IRQ eth_wake_irq not found
[    1.329032] stmmaceth 18140000.ethernet: IRQ eth_lpi not found
[    1.335852] stmmaceth 18140000.ethernet: PTP uses main clock
[    1.342734] stmmaceth 18140000.ethernet: User ID: 0x10, Synopsys ID: 0x37
[    1.350407] stmmaceth 18140000.ethernet:     DWMAC1000
[    1.355911] stmmaceth 18140000.ethernet: DMA HW capability register supported
[    1.363868] stmmaceth 18140000.ethernet: RX Checksum Offload Engine supported
[    1.371845] stmmaceth 18140000.ethernet: COE Type 2
[    1.377318] stmmaceth 18140000.ethernet: TX Checksum insertion supported
[    1.384789] stmmaceth 18140000.ethernet: Wake-Up On Lan supported
[    1.391606] stmmaceth 18140000.ethernet: Normal descriptors
[    1.397852] stmmaceth 18140000.ethernet: Ring mode enabled
[    1.403971] stmmaceth 18140000.ethernet: Enable RX Mitigation via HW Watchdog Timer
[    1.422456] Micrel KSZ8081 or KSZ8091 stmmac-0:00: attached PHY driver (mii_bus:phy_addr=stmmac-0:00, irq=POLL)
[    1.438845] dwc2 18120000.usb: supply vusb_d not found, using dummy regulator
[    1.447406] dwc2 18120000.usb: supply vusb_a not found, using dummy regulator
[    1.595390] dwc2 18120000.usb: dwc2_check_param_tx_fifo_sizes: Invalid parameter g_tx_fifo_size[5]=0
[    1.605633] dwc2 18120000.usb: dwc2_check_param_tx_fifo_sizes: Invalid parameter g_tx_fifo_size[6]=0
[    1.615843] dwc2 18120000.usb: dwc2_check_param_tx_fifo_sizes: Invalid parameter g_tx_fifo_size[7]=0
[    1.626052] dwc2 18120000.usb: dwc2_check_param_tx_fifo_sizes: Invalid parameter g_tx_fifo_size[8]=0
[    1.636336] dwc2 18120000.usb: EPs: 9, dedicated fifos, 4664 entries in SPRAM
[    1.645006] dwc2 18120000.usb: DWC OTG Controller
[    1.650337] dwc2 18120000.usb: new USB bus registered, assigned bus number 1
[    1.658265] dwc2 18120000.usb: irq 32, io mem 0x18120000
[    1.664589] usb usb1: New USB device found, idVendor=1d6b, idProduct=0002, bcdDevice= 5.15
[    1.673840] usb usb1: New USB device strings: Mfr=3, Product=2, SerialNumber=1
[    1.681924] usb usb1: Product: DWC OTG Controller
[    1.687207] usb usb1: Manufacturer: Linux 5.15.89 dwc2_hsotg
[    1.693517] usb usb1: SerialNumber: 18120000.usb
[    1.699486] hub 1-0:1.0: USB hub found
[    1.703793] hub 1-0:1.0: 1 port detected
[    1.710246] ehci_hcd: USB 2.0 'Enhanced' Host Controller (EHCI) Driver
[    1.717904] usbcore: registered new interface driver usb-storage
[    1.728574] imgpdc-wdt 18102100.watchdog: watchdog module last reset due to soft reset
[    1.738945] Synopsys Designware Multimedia Card Interface Driver
[    1.746933] dw_mmc 18142000.mmc: IDMAC supports 32-bit address mode.
[    1.750447] NET: Registered PF_INET6 protocol family
[    1.754103] dw_mmc 18142000.mmc: Using internal DMA controller.
[    1.766309] dw_mmc 18142000.mmc: Version ID is 270a
[    1.767721] Segment Routing with IPv6
[    1.771993] dw_mmc 18142000.mmc: DW MMC controller at irq 34,32 bit host data width,32 deep fifo
[    1.776045] In-situ OAM (IOAM) with IPv6
[    1.790506] NET: Registered PF_PACKET protocol family
[    1.796466] 8021q: 802.1Q VLAN Support v1.8
[    1.799280] mmc_host mmc0: Bus speed (slot 0) = 87500000Hz (slot req 400000Hz, actual 397727HZ div = 110)
[    1.836596] ubi0: default fastmap pool size: 50
[    1.841781] ubi0: default fastmap WL pool size: 25
[    1.847166] ubi0: attaching mtd4
[    2.315398] random: crng init done
[    3.927168] ubi0: scanning is finished
[    3.947309] ubi0 warning: ubi_eba_init: cannot reserve enough PEBs for bad PEB handling, reserved 38, need 40
[    3.960500] ubi0: attached mtd4 (name "firmware0", size 256 MiB)
[    3.967299] ubi0: PEB size: 262144 bytes (256 KiB), LEB size: 253952 bytes
[    3.974970] ubi0: min./max. I/O unit sizes: 4096/4096, sub-page size 4096
[    3.982559] ubi0: VID header offset: 4096 (aligned 4096), data offset: 8192
[    3.990342] ubi0: good PEBs: 1024, bad PEBs: 0, corrupted PEBs: 0
[    3.997156] ubi0: user volume: 3, internal volumes: 1, max. volumes count: 128
[    4.005210] ubi0: max/mean erase counter: 89/45, WL threshold: 4096, image sequence number: 516423239
[    4.015501] ubi0: available PEBs: 0, total reserved PEBs: 1024, PEBs reserved for bad PEB handling: 38
[    4.026059] ubi0: background thread "ubi_bgt0d" started, PID 304
[    4.035640] block ubiblock0_1: created from ubi0:1(rootfs)
[    4.041816] ubiblock: device ubiblock0_1 (rootfs) set to be root filesystem
[    4.101536] VFS: Mounted root (squashfs filesystem) readonly on device 254:0.
[    4.110543] Freeing unused kernel image (initmem) memory: 244K
[    4.117091] This architecture does not have kernel memory protection.
[    4.124306] Run /sbin/init as init process
[    4.683074] init: Console is alive
[    4.687475] init: - watchdog -
[    5.138482] kmodloader: loading kernel modules from /etc/modules-boot.d/*
[    5.203749] tpm_i2c_infineon 0-0020: 1.2 TPM (device-id 0x1A)
[    5.270525] tpm tpm0: TPM is disabled/deactivated (0x7)
[    5.295643] kmodloader: done loading kernel modules from /etc/modules-boot.d/*
[    5.310298] init: - preinit -
[    6.534365] stmmaceth 18140000.ethernet eth0: Register MEM_TYPE_PAGE_POOL RxQ-0
[    6.639098] stmmaceth 18140000.ethernet eth0: PHY [stmmac-0:00] driver [Micrel KSZ8081 or KSZ8091] (irq=POLL)
[    6.655332] stmmaceth 18140000.ethernet eth0: No Safety Features support found
[    6.667442] stmmaceth 18140000.ethernet eth0: IEEE 1588-2002 Timestamp supported
[    6.676111] stmmaceth 18140000.ethernet eth0: registered PTP clock
[    6.685225] stmmaceth 18140000.ethernet eth0: configuring for phy/rmii link mode
Press the [f] key and hit [enter] to enter failsafe mode
Press the [1], [2], [3] or [4] key and hit [enter] to select the debug level
[    8.905879] UBIFS (ubi0:2): Mounting in unauthenticated mode
[    8.912621] UBIFS (ubi0:2): background thread "ubifs_bgt0_2" started, PID 422
[    9.048729] UBIFS (ubi0:2): recovery needed
[    9.342287] UBIFS (ubi0:2): recovery completed
[    9.347517] UBIFS (ubi0:2): UBIFS: mounted UBI device 0, volume 2, name "rootfs_data"
[    9.356275] UBIFS (ubi0:2): LEB size: 253952 bytes (248 KiB), min./max. I/O unit sizes: 4096 bytes/4096 bytes
[    9.367353] UBIFS (ubi0:2): FS size: 238460928 bytes (227 MiB, 939 LEBs), max 950 LEBs, journal size 11935744 bytes (11 MiB, 47 LEBs)
[    9.380762] UBIFS (ubi0:2): reserved for root: 4952683 bytes (4836 KiB)
[    9.388161] UBIFS (ubi0:2): media format: w5/r0 (latest is w5/r0), UUID DD8741CE-2FFE-4925-91F6-2A7B13846E26, small LPT model
[    9.412943] mount_root: switching to ubifs overlay
[    9.449753] urandom-seed: Seeding with /etc/urandom.seed
[    9.592990] procd: - early -
[    9.596703] procd: - watchdog -
[   10.215554] procd: - watchdog -
[   10.220061] procd: - ubus -
[   10.390968] procd: - init -
Please press Enter to activate this console.
[   11.319030] kmodloader: loading kernel modules from /etc/modules.d/*
[   11.331985] i2c_dev: i2c /dev entries driver
[   11.344898] NET: Registered PF_IEEE802154 protocol family
[   11.362967] ca8210 spi0.0: Inserting ca8210
[   11.529965] ca8210 spi0.0: Wakeup indication received, reason:
[   11.536505] ca8210 spi0.0: Transceiver woken up from Power Up / System Reset
[   11.557612] ca8210 spi0.0: Switching external clock on
[   11.565047] ca8210 spi0.0: External clock set as clock provider
[   11.656127] urngd: v1.0.2 started.
[   11.660554] PPP generic driver version 2.4.2
[   11.667895] NET: Registered PF_PPPOX protocol family
[   11.686219] kmodloader: done loading kernel modules from /etc/modules.d/*
[   19.528146] stmmaceth 18140000.ethernet eth0: Register MEM_TYPE_PAGE_POOL RxQ-0
[   19.619540] stmmaceth 18140000.ethernet eth0: PHY [stmmac-0:00] driver [Micrel KSZ8081 or KSZ8091] (irq=POLL)
[   19.635489] stmmaceth 18140000.ethernet eth0: No Safety Features support found
[   19.648533] stmmaceth 18140000.ethernet eth0: IEEE 1588-2002 Timestamp supported
[   19.657656] stmmaceth 18140000.ethernet eth0: registered PTP clock
[   19.664730] stmmaceth 18140000.ethernet eth0: configuring for phy/rmii link mode
[   19.678015] br-lan: port 1(eth0) entered blocking state
[   19.683886] br-lan: port 1(eth0) entered disabled state
[   19.690502] device eth0 entered promiscuous mode
[   22.796076] stmmaceth 18140000.ethernet eth0: Link is Up - 100Mbps/Full - flow control rx/tx
[   22.805700] br-lan: port 1(eth0) entered blocking state
[   22.811539] br-lan: port 1(eth0) entered forwarding state
[   22.819277] IPv6: ADDRCONF(NETDEV_CHANGE): br-lan: link becomes ready
[   34.485345] aux_adc_vref: disabling
[   34.489327] internal_dac_supply: disabling



BusyBox v1.36.0 (2023-01-22 16:45:41 UTC) built-in shell (ash)

  _______                     ________        __
 |       |.-----.-----.-----.|  |  |  |.----.|  |_
 |   -   ||  _  |  -__|     ||  |  |  ||   _||   _|
 |_______||   __|_____|__|__||________||__|  |____|
          |__| W I R E L E S S   F R E E D O M
 -----------------------------------------------------
 OpenWrt SNAPSHOT, r21887+4-fb7f4d4b549f
 -----------------------------------------------------
=== WARNING! =====================================
There is no root password defined on this device!
Use the "passwd" command to set up a new password
in order to prevent unauthorized SSH logins.
--------------------------------------------------
root@OpenWrt:/# uname -a
Linux OpenWrt 5.15.89 #0 SMP Sun Jan 22 16:45:41 2023 mips GNU/Linux
root@OpenWrt:/# echo "mtd2: 2000 1000 \"uEnv\"" | awk -F: '{print $1}'
mtd2: 2000 1000 "uEnv"
root@OpenWrt:/# 
````

Could you please create a separate pull request for the last patch. 
I also have a patch to clean up the kernel configuration a bit more. I will create a separate pull request later.

PolynomialDivision committed:
pistachio: switch to 5.15 by default

It appears that only a few users are using the pistachio SoC. The most
active user of the target has already approved the testing kernel and
so it is very unlikely bugs will be reported in the near future.
Therefore, the target should be directly bumped to 5.15.

Acked-by: Hauke Mehrtens <hauke@hauke-m.de>
Signed-off-by: Nick Hainke <vincent@systemli.org>

PolynomialDivision commented:
@hauke Thanks! I added you as co-devloped and tested by. I also added a commit changing directly the default kernel config.
